amps/ags/rm-community/rm-community-repo/unit-test/java/org/alfresco/module/org_alfresco_module_rm/disposition/RetentionScheduleModelUnitTest.java

```diff
@@ -0,0 +1,110 @@
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ * -
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * -
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ * -
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.module.org_alfresco_module_rm.disposition;
+
+import org.alfresco.module.org_alfresco_module_rm.model.RecordsManagementModel;
+import org.alfresco.module.org_alfresco_module_rm.test.util.BaseUnitTest;
+import org.alfresco.rm.rest.api.impl.ApiNodesModelFactory;
+import org.alfresco.rm.rest.api.model.RetentionSchedule;
+import org.alfresco.rm.rest.api.model.RetentionScheduleActionDefinition;
+import org.alfresco.service.cmr.repository.ChildAssociationRef;
+import org.alfresco.service.cmr.repository.NodeRef;
+import org.alfresco.service.cmr.repository.Period;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.when;
+
+/**
+ * Retention schedule model unit test
+ */
+public class RetentionScheduleModelUnitTest extends BaseUnitTest
+{
+    private static final String AUTHORITY = "authority";
+    private static final String INSTRUCTIONS = "instructions";
+    private static final String RETAIN_STEP = "retain";
+
+    @InjectMocks
+    private ApiNodesModelFactory apiNodesModelFactory;
+
+    @Mock
+    DispositionSchedule dispositionSchedule;
+
+    @Mock
+    DispositionActionDefinition dispositionActionDefinition;
+
+    @Test
+    public void mapRetentionScheduleDataTest()
+    {
+        // Mock data
+        NodeRef nodeRef = generateNodeRef(RecordsManagementModel.TYPE_DISPOSITION_SCHEDULE, true);
+        ChildAssociationRef childAssociationRef = generateChildAssociationRef(filePlan, record);
+        when(dispositionSchedule.getDispositionAuthority()).thenReturn(AUTHORITY);
+        when(dispositionSchedule.getDispositionInstructions()).thenReturn(INSTRUCTIONS);
+        when(dispositionSchedule.getNodeRef()).thenReturn(nodeRef);
+        when(dispositionSchedule.isRecordLevelDisposition()).thenReturn(false);
+        when(mockedNodeService.getPrimaryParent(nodeRef)).thenReturn(childAssociationRef);
+        // Call the method
+        RetentionSchedule expectedResult = apiNodesModelFactory.mapRetentionScheduleData(dispositionSchedule);
```

Comment:
Here you have mislabelled the actual result as "expected" and you're still making a list of individual assertions rather than relying on the `equals` method of `RetentionSchedule`.